### PR TITLE
Add ID migration tool

### PR DESCRIPTION
**Pull request by byliuyang**
### Description
Previously we are migrating from email to user ID. However, after the first and the second DB migrations, we unfortunately lost the most recent user data and had to redo the migration.

### Description
Add ID migration tool so that we can re-perform the migration.

### CODE REVIEW

File: backend/cmd/cmd.go [hunk 1]
```diff
@@ -41,6 +43,23 @@ func NewRootCmd(
 		"migration migrations root directory",
 	)
 
+	idCmd := cmdFactory.NewCommand(cli.CommandConfig{
+		Usage:        "data-id",
+		ShortHelpMsg: "Use user ID to uniquely identify a user",
+		OnExecute: func(cmd *cli.Command, args []string) {
```
Comment (arberiii) [inline backend/cmd/cmd.go:49]: cli.Command is interface therefore a pointer itself

File: backend/cmd/cmd.go [hunk 2]
```diff
@@ -41,6 +43,23 @@ func NewRootCmd(
...
+			kgsConfig := provider.KgsRPCConfig{
+				Hostname: config.KgsHostname,
+				Port:     config.KgsPort,
+			}
+			keyGenBufferSize := provider.KeyGenBufferSize(config.KeyGenBufferSize)
+			dataTool, err := dep.InjectDataTool(dbConfig, dbConnector, keyGenBufferSize, kgsConfig)
+			if err != nil {
+				panic(err)
```
Comment (arberiii) [inline backend/cmd/cmd.go:57]:
```suggestion
fmt.Println(err)
os.Exit(1)
```

Comment (byliuyang) [reply backend/cmd/cmd.go:57]: @arberiii Why not panic? I think it's okey to crash if we running the tool offline.

File: backend/cmd/cmd.go (hunk 1 above)
Comment (byliuyang) [reply backend/cmd/cmd.go:49]: @arberiii Yes?

File: backend/cmd/cmd.go (hunk 2 above)
Comment (arberiii) [reply backend/cmd/cmd.go:57]: I thought to keep the same error handling through out the function

File: backend/cmd/cmd.go (hunk 1 above)
Comment (arberiii) [reply backend/cmd/cmd.go:49]: I thought we don’t to pass the pointer, like just cli.Command. What do you think?

File: backend/cmd/cmd.go (hunk 2 above)
Comment (byliuyang) [reply backend/cmd/cmd.go:57]: Done.

File: backend/cmd/cmd.go (hunk 1 above)
Comment (byliuyang) [reply backend/cmd/cmd.go:49]: @arberiii Adde TODO to update the signature in the app framework.

File: backend/tool/data.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+// Data transform existing data to the new format and move them to the correct
+// location.
+type Data struct {
+	keyGen keygen.KeyGenerator
+	db     *sql.DB
+}
+
+// EmailToID generates IDs for users and changes DB tables reference those IDs.
+func (d Data) EmailToID() {
+	rows, err := d.db.Query(`
```
Comment (alldroll) [inline backend/tool/data.go:19]: I suggest using batch reading

```
SELECT email FROM "user" where "id is empty" LIMIT 1000
```

Here "id is empty" means, that we fetch only records with unsettled `id`. We perform the above query again and again until after the select dataset size is less or equal to 1000.

It gives us a feeling to be sure that we can handle this migration even for a big number of records "just a good practice"

File: backend/cmd/cmd.go (hunk 2 above)
Comment (alldroll) [reply backend/cmd/cmd.go:57]: I think we should `fmt.Error` with a `%w` verb.

File: backend/tool/data.go (hunk 3 above)
Comment (byliuyang) [reply backend/tool/data.go:19]: @alldroll Should we automatically increase the number of counts using for loops or we give admin the control to decide the size?

Comment (alldroll) [reply backend/tool/data.go:19]: It depends but in general, we increase the number of counts. I think the simplest way is the best one:)

File: backend/cmd/cmd.go (hunk 2 above)
Comment (byliuyang) [reply backend/cmd/cmd.go]: @alldroll Wa are directly printing the error here. 'fmt.Errorf' returns unused result and didn't provide much value in this case. I think it's better to use `fmt.Errorf` inside the application logic.

File: backend/tool/data.go (hunk 3 above)
Comment (byliuyang) [reply backend/tool/data.go:24]: @alldroll Done.

**Pull request merged**